### DISCUSSION TITLE
Wrong outputs from ru_core_news_sm model with hyphened words

### DISCUSSION THREAD

**Discussion by riZZZhik** [explosion/spaCy#10605] [Help: Coding & Implementations] [answered]
Hi, got strange outputs on lemmatizing Russian words with hyphen:

Initialization:
```python
import spacy
nlp = spacy.load('ru_core_news_sm')
```

Test 1:
```python
print([(x.lemma, x.lemma_) for x in nlp('по-любому')]
```
Output: `[(12047934663327436226, 'по'), (9153284864653046197, '-'), (1520167342668405783, 'любому')]`

Test 2:
```python
print([(x.lemma, x.lemma_) for x in nlp('любому')]
```
Output: `[(11783268970549905777, 'любой')]`

Question:
Why the word 'любому' in second test lemmatized as 'любой' - what is correct, but in first as 'любому'?

**Comment by thomashacker** [comment, marked as answer]
Hello,
the lemmatizer (`pymorphy2`) depends on POS annotations from the tagger. I'd suggest checking the POS tags first, maybe it makes it more clear why the lemmatization breaks in your first example.

**Reply by riZZZhik** [reply]
Yeah, different POS tags, I will research about them and return if there would be any other questions.
Thank you very much!

```python
[(x.pos_, x.lemma_) for x in nlp('по-любому')] # [('ADV', 'по'), ('ADV', '-'), ('ADV', 'любому')]
[(x.pos_, x.lemma_) for x in nlp('любому')]    # [('DET', 'любой')]
```
